LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan/NetId.cs

```diff
@@ -31,5 +31,36 @@ public NetId(int value)
         public int NetworkId => this.value & 0b111_1111;
 
         public override string ToString() => this.value.ToString("X6", CultureInfo.InvariantCulture);
+
+        public static NetId Parse(ReadOnlySpan<char> input) =>
+            TryParse(input, out var result) ? result : throw new FormatException();
+
+        public static bool TryParse(ReadOnlySpan<char> input, out NetId result)
+        {
+            if (int.TryParse(input, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var raw))
```

Comment:
`raw` is an `int` but a **NetID** is supposed to be a 24-bit number. If `raw` is too big then `NetId` will throw `ArgumentOutOfRangeException`, which shouldn't happen for a method that's meant to _try and parse_. I would add a test to validate that and the fix should be as follows:

```suggestion
            if (int.TryParse(input, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var raw)
                && unchecked((uint)raw & 0xff00_0000) == 0)
```

You could then introduce a helper static method in the type along the lines of:

```c#
private static bool IsSesquiWord(int n) => unchecked((uint)n & 0xff00_0000) == 0;
```

so that the constructor and this method use a single validation logic.

Reply:
Applied with 892492d5f0db18378c49299e9601f658fcf3866c